### DISCUSSION TITLE
If shift_state=D and geofence=home then open the gate of house

### DISCUSSION THREAD

**Discussion by denisjoshua** [teslamate-org/teslamate#1811] [Q&A]
Hi there all, 
I just make a Node_Red script that open the gate if the zone of Tesla is Home 
and the shift_state go in "D" mode. 
The problem is that sometime is work sometime is not work. 

I don't understand why TeslaMate sometime is recognize it and sometime is not. 
Can someone help me please ? 

Thanks in advance 
Denis

**Comment by LelandSindt** [comment]
What you are trying to do sounds similar to the logic I use for [retracor](https://github.com/adriankumpf/teslamate/discussions/1440). 

Have you enabled "Steaming API"?

Is `state` = `suspended`  when you see the failure?

For me information provided by TeslaMate via MQTT has been reliable and timely,

**Reply by denisjoshua** [reply]
The state is "online" ... but I understand now that the shift_state some time will arrive after 40 seconds. 
I just make about 5 test... sometime shift_state arrive in about 2 seconds sometime in more than 40 seconds. 

Anyway I'll let like this for now. In case I forget the remote I prefer to wait 40 seconds :-) 

And just for fun here is my Node-Red flow for now: 
![image](https://user-images.githubusercontent.com/32936800/125094018-32cdc700-e0d3-11eb-8a1c-926d3fdcfdd3.png)

This manage the PV system, Tesla PowerWall battery and the Tesla M3 charge :-) 

Thanks for the help 
Denis
